### DISCUSSION TITLE
[com_fields] Field Input Class doesn't work

### DISCUSSION THREAD

**Discussion by AndySDH** [joomla/joomla-cms#47176] [2. Cold Features]
### Steps to reproduce the issue

Create a custom field.

Give a "Field Class". 
![image](https://user-images.githubusercontent.com/25923699/69709316-77267200-10fd-11ea-92a0-fbf4f0e8f512.png)

### Expected result

This class is supposed to be added to the control-group of the field in the edit form.

### Actual result

It's not added anywhere.

### System information (as much as possible)

Joomla 3.9.3

**Comment by Quy** [comment]
They are added. See `LabelCustomClass` and `FieldCustomClass`.

```
<div id="attrib-fields-0" class="tab-pane">
<div class="control-group">
			<div class="control-label"><label id="jform_com_fields_text-lbl" for="jform_com_fields_text" class="LabelCustomClass">
	Text</label>
</div>
		<div class="controls"><input type="text" name="jform[com_fields][text]" id="jform_com_fields_text"  value="" class="FieldCustomClass"              />
</div>
</div>
```

**Comment by AndySDH** [comment]
Oh they're added to the input, not to the main control-group. That's not very useful. It should really be added to the control-group to be able to style the field. Not much you can style when adding it to the input.

**Comment by Quy** [comment]
Please provide an example that you would you do that you can't do now as it.

**Comment by mbabker** [comment]
A wrapping container (`.control-group` or `.form-group` with Bootstrap, `.field` with Semantic UI, etc. etc.) is really more of a framework specific design.  To me it's right having the field class apply to the actual form field element (`<input>`, `<select>`, `<textarea>`) and not a wrapping container.  And I'd be hesitant to suggest a param for a class on a wrapping container.  Then again, the Bootstrap coupling is so real that maybe it isn't that terrible of a thing to force.

**Comment by ciar4n** [comment]
A similar issue here https://github.com/joomla/joomla-cms/issues/11887

I agree with @AndySDH on this. True, if you rely completely on CSS frameworks then a class on the field itself has some benefit. For those that write their own CSS, a class on the parent `.control-group` gives you far greater scope in what can be achieved.

**Comment by AndySDH** [comment]
> Please provide an example that you would you do that you can't do now as it.

Well it's very simple. With the current way, you can only style the input.

`<input type="text" name="jform[com_fields][text]" id="jform_com_fields_text" class="FieldCustomClass"  />`

Instead, if the class was in the control-group:

`<div class="control-group" class="FieldCustomClass">....</div>`

You would be able to style the entire field output, give it a custom background, a custom border, or whatever else. And still also be able to style the input with` .FieldCustomClass input`

So I see no side effects in having the class in the control-group. It's kinda useless just in the input imho. Didn't even notice it was there and let me to believe it was broken.

**Comment by ciar4n** [comment]
From a user perspective, there is framework classes, they may wish to apply directly to a field which is the point @mbabker is making. It really depends on the type of user we expect with Joomla.

**Comment by joomdonation** [comment]
@AndySDH Is this still needed? Or we can live with what we are having now ?

**Comment by AndySDH** [comment]
I would say this is still needed. It would be a significant improvement and would allow to style individual fields in the form, which is not possible now.

**Comment by joomdonation** [comment]
OK. Thanks for your feedback. Hopefully someone will come up with a solution (I haven't spent enough time with custom fields yet, so I could not comment about it further)

**Comment by fancyFranci** [comment]
Would a third field "wrapper class" could be a solution?

**Comment by no64** [comment]
We have now reached Joomla 6, yet the class for the fields is still not being applied—at least not for "media fields" or newer types like "audio," "video," or "document."

In my opinion, fields should have their own dedicated class by default so they can be targeted via CSS. Currently, the designations are merely "controls," "control-groups," and so on. If there were a class for the entire field, it might be possible to hide some of the "countless" field labels; these are particularly bothersome when using "subform" fields, as a new label appears with every level of nesting.

Furthermore, when using these fields for "dynamic content" in conjunction with page builders like "YOOtheme," you might not even need additional input fields, such as those for "alt text" in the 'media field'.

Overall, this would allow for a cleaner layout, free from distracting, repetitive elements.
